### PR TITLE
Draft of user and implementer guidelines

### PR DESCRIPTION
**Pull request by syg**
Feedback welcome

### CODE REVIEW

**Comment by syg** [comment]
cc @codehag and @kmiller68

File: spec.html [hunk 1]
```diff
@@ -942,6 +942,30 @@ <h1>get DataView.prototype.byteOffset</h1>
...
+  </emu-note>
+
+    <emu-note>
+      <p>The following are guidelines for ECMAScript implementers implementing GrowableSharedArrayBuffer.</p>
+      <p>We recommend GrowableSharedArrayBuffer be implemented as in-place growth via reserving virtual memory up front.</p>
+      <p>Because grow operations can be concurrent with memory accesses on a GrowableSharedArrayBuffer, the constraints of the memory model require that even unordered accesses do not "tear" (bits of their values will not be mixed). In practice, this means the underlying data block of a GrowableSharedArrayBuffer cannot be grown by being copied without stopping the world. We do not recommend stopping the world as an implementation strategy because it introduces a serialization point and its slowness.</p>
+      <p>If a host does not have virtual memory, such as those running on embedded devices, we recommend ignoring the _requestedMaxByteLength_ parameter, throwing on calling GorwableSharedArrayBuffer.prototype.grow, and communicating to users that GrowableSharedArrayBuffers cannot be grown.</p>
```
Comment (ljharb) [inline spec.html]:
```suggestion
      <p>If a host does not have virtual memory, such as those running on embedded devices, we recommend ignoring the _requestedMaxByteLength_ parameter, throwing on calling GrowableSharedArrayBuffer.prototype.grow, and communicating to users that GrowableSharedArrayBuffers cannot be grown.</p>
```

**Review by ljharb — COMMENTED** [review]
when did "GrowableArrayBuffer" become a thing?

**Comment by syg** [comment]
> when did "GrowableArrayBuffer" become a thing?

There's no GrowableArrayBuffers, only GrowableSharedArrayBuffers, and that's been part of this proposal from the beginning. The point is that SABs can't shrink because of concurrency concerns, while ABs can both grow and shrink.

**Comment by ljharb** [comment]
ah, thanks for clarifying.

**Pull request merged**